halo2-ecc/src/bn254/bls_signature.rs

```diff
@@ -41,55 +41,17 @@ impl<'chip, F: PrimeField> BlsSignatureChip<'chip, F> {
         let fp2_chip = Fp2Chip::<F>::new(self.fp_chip);
         let g2_chip = EccChip::new(&fp2_chip);
 
-        let g1_assigned = self.pairing_chip.load_private_g1_unchecked(ctx, g1);
-        // Checking element from G1 is on curve also check that it's in subgroup G1 since G1 has cofactor of 1
-        g1_chip.assert_is_on_curve::<halo2_base::halo2_proofs::halo2curves::bn256::G1Affine>(
-            ctx,
-            &g1_assigned,
-        );
-
-        let hash_m_assigned = self.pairing_chip.load_private_g2_unchecked(ctx, msghash);
-        g2_chip.assert_is_on_curve::<halo2_base::halo2_proofs::halo2curves::bn256::G2Affine>(
-            ctx,
-            &hash_m_assigned,
-        );
+        let g1_assigned = self.pairing_chip.load_private_g1(ctx, g1);
 
-        let mut signature_agg_assigned =
-            self.pairing_chip.load_private_g2_unchecked(ctx, signatures[0]);
-        g2_chip.field_chip.enforce_less_than(ctx, signature_agg_assigned.x().clone());
-        let mut pubkey_agg_assigned = self.pairing_chip.load_private_g1_unchecked(ctx, pubkeys[0]);
-        g1_chip.field_chip.enforce_less_than(ctx, pubkey_agg_assigned.x().clone());
+        let hash_m_assigned = self.pairing_chip.load_private_g2(ctx, msghash);
 
-        // loop through signatures and aggregate them
-        for (index, signature) in signatures.iter().enumerate() {
-            if index > 0 {
-                let signature_assigned =
-                    self.pairing_chip.load_private_g2_unchecked(ctx, *signature);
-                g2_chip.field_chip.enforce_less_than(ctx, signature_assigned.x().clone());
-                g2_chip
-                    .assert_is_on_curve::<halo2_base::halo2_proofs::halo2curves::bn256::G2Affine>(
-                        ctx,
-                        &signature_assigned,
-                    );
-                signature_agg_assigned =
-                    g2_chip.add_unequal(ctx, &signature_agg_assigned, &signature_assigned, false);
-            }
-        }
+        let signature_points =
+            signatures.iter().map(|pt| g2_chip.assign_point(ctx, *pt)).collect::<Vec<_>>();
```

Comment:
This is a nuance (very hard for you to be able to distinguish), but `assign_point` allows (0,0) for point at infinity, while `load_private` does not. I'm a little scared about (0,0), especially for signatures, so let's just use `load_private` for now.

We'll try to do a comprehensive refactor to handle (0,0) more systematically later.

Reply:
I saw the the comments of the 2 respective functions and I wasn't sure if infinity will have any implications. Thanks for pointing this out!